README.md

```diff
@@ -119,6 +119,184 @@ See github documentation for more information:
 
 NOTE: an alternative option is to install and use the [hub cli tool](https://github.com/github/hub) which provides a convenient way to generate and save github access token.
 
+### Large Tests
+
+Large test requires the following software on your development systems:
+
+* docker
+* docker-compose
+
+The default large test performs the following actions:
+
+* use environment variable to populate and initialize containers in: `scripts/test/docker_compose.yml`
+* conditionally run `scripts/test/setup.rb` before any test (use this to create test database, test service etc)
+* download and run the appropriate version of Snap per `$SNAP_VERSION`
+* scan `examples/task/*.yml` for list of tasks and metrics
+* load snap plugins, first from the local `build/linux/x86_64/*` directory, try `build.snap-telemetry.io` s3 bucket next
+* verify plugins are loaded successfully
+* attempt to create, verify, and stop every yaml task in the examples directory.
+* shutdown and cleanup containers
+
+If this is not the appropriate behavior, you can write custom large test as `{test_name}_spec.rb` in the `scripts/test` directory.
+
+#### Layout
+
+Pluginsync adds several files to support large test framework. Files names surrounded by {} are not created by pluginsync. Please review the comment to see how each file affects large tests.
+
+```
+├── build
+│   └── linux
+│       └── x86_64                                  # plugins missing from this directory will be fetched from s3
+│           ├── {snap-plugin-collector-plugin_name} # make will build the current plugin for testing
+│           └── {snap-plugin-publisher-custom}      # custom plugins can be copied here to take precedence over s3 binaries
```

Comment:
`snap-plugin-publisher-custom` can be confusing too.

What do you think about `snap-plugin-type-custom`?